dask/dataframe/tests/test_reshape.py

```diff
@@ -191,6 +191,32 @@ def test_pivot_table(values, aggfunc):
     assert_eq(res, exp)
 
 
+@pytest.mark.parametrize("values", ["B", ["D"], ["B", "D"]])
+@pytest.mark.parametrize("aggfunc", ["first", "last"])
+def test_pivot_table_firstlast(values, aggfunc):
+
+    df = pd.DataFrame(
+        {
+            "A": np.random.choice(list("XYZ"), size=100),
+            "B": np.random.randn(100),
+            "C": pd.Categorical(np.random.choice(list("abc"), size=100)),
+            "D": np.random.choice(list("abc"), size=100),
+        }
+    )
+    ddf = dd.from_pandas(df, 5).repartition((0, 20, 40, 60, 80, 98, 99))
+
+    res = dd.pivot_table(ddf, index="A", columns="C", values=values, aggfunc=aggfunc)
+    exp = pd.pivot_table(df, index="A", columns="C", values=values, aggfunc=aggfunc)
+
+    assert exp.equals(res.compute())
```

Comment:
It's better to use `assert_eq`, as it also checks that the metadata on the dask object (e.g. columns, dtypes, ...) matches the computed result.

```suggestion
    assert_eq(res, exp)
```

Reply:
Thanks for the comment, I see your point. The reason I chose this approach was to make test for `first` and `last` aggregate functions work with non-numeric data, i.e. strings. In 'pivot_table' the data type is set explicitly to `float64`:
``
meta = pd.DataFrame(
        columns=new_columns, dtype=np.float64, index=pd.Index(df._meta[index])
    )
``
Maybe there is need for a more fundamental change to make it work with non-numeric data?

Reply:
I agree. I have removed my test_pivot_table_firstlast function and only use the existing test_pivot_table function.